### PR TITLE
[Native] Fix PartitionedOutputNode plan node id

### LINKED ISSUE #21741: [native] Output operators are missing in Stage Summary Page
## Current Behavior
In same production query
CPP VS Java
image

## Possible Reason

From debugging front end
image

The output operator should have the same planNodeId as its parent, [jsx logic ](https://github.com/prestodb/presto/blob/f4f434a243a3d126738f697415bd39b391232fa5/presto-main/src/main/resources/webapp/src/components/StageDetail.jsx#L390)
CPP's PartitionedOutput's planNodeId is **"root"**

CPP
```
"stageId": 1,
 "stageExecutionId": 0,
"pipelineId": 0,
"operatorId": 1,
"planNodeId": "185",
 "operatorType": "PartialAggregation",

"stageId": 1,
"stageExecutionId": 0,
"pipelineId": 0,
"operatorId": 2,
"planNodeId": "root",
"operatorType": "PartitionedOutput",
```
JAVA
```
"stageId": 1,
"stageExecutionId": 0,
"pipelineId": 0,
"operatorId": 1,
"planNodeId": "185",
"operatorType": "AggregationOperator",

"stageId": 1,
"stageExecutionId": 0,
"pipelineId": 0,
"operatorId": 2,
"planNodeId": "185",
"operatorType": "TaskOutputOperator",
```

### PR DESCRIPTION
**Pull request by kewang1024**
Presto doesn't have PartitionedOutputNode and assigns its source node's plan
node id to PartitionedOutputOperator.
so Presto UI is expecting PartitionedOutputOperator having same node id as its 
source operator.
However, Velox has PartitionedOutputNode whose plan node is is set to "root". 

To comply with original assumption, fix partitionedOutputNode plan node id to be
`root.<source plan node ID>` and parse them back to source plan node ID in task 
stats reporting.

Fixes #21741

### CODE REVIEW

**Comment by kewang1024** [comment]
Steps to Reproduce
Set up both environments locally:

Presto HiveQueryRunner [1 worker]
Prestissimo environment (HiveExternalWorkerQueryRunner) [1 worker]
Ran query in both environments:
insert into orders_ke select orderkey, custkey, orderstatus, totalprice, '2011-01-01', '1', 'clerk', 2, 'comment' from orders;

In Java cluster UI

image

File: presto-native-execution/presto_cpp/main/PrestoTask.cpp [hunk 1]
```diff
@@ -77,6 +77,16 @@ protocol::RuntimeUnit toPrestoRuntimeUnit(RuntimeCounter::Unit unit) {
   }
 }
 
+// Presto operator's node id sometimes is not equivalent to velox's.
+// So when reporting task stats, we need to parse node id back to presto's.
+// For example, velox's partitionedOutput operator would have "root." prefix.
+std::string toPrestoPlanNodeId(const protocol::PlanNodeId& id) {
+  if (id.find("root.") != std::string::npos) {
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/PrestoTask.cpp:84]:
```
if (FOLLY_LIKELY) {
  return id;
}
return id.substr(5);
```

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 2]
```diff
@@ -2768,6 +2768,16 @@ core::ExecutionStrategy toStrategy(protocol::StageExecutionStrategy strategy) {
   }
   VELOX_UNSUPPORTED("Unknown Stage Execution Strategy type {}", (int)strategy);
 }
+
+// Presto doesn't have PartitionedOutputNode and assigns its source node's plan
+// node id to PartitionedOutputOperator.
+// However, Velox has PartitionedOutputNode and doesn't allow duplicate node
+// ids, so we use "root." + source node's plan id as PartitionedOutputNode's.
+// For example "root.10" if source node's plan node is "10".
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:2776]:
```
For example, if source node plan id is "10", then the associated partitioned output node id is "root.10".
```

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 3]
```diff
@@ -2768,6 +2768,16 @@ core::ExecutionStrategy toStrategy(protocol::StageExecutionStrategy strategy) {
   }
   VELOX_UNSUPPORTED("Unknown Stage Execution Strategy type {}", (int)strategy);
 }
+
+// Presto doesn't have PartitionedOutputNode and assigns its source node's plan
+// node id to PartitionedOutputOperator.
+// However, Velox has PartitionedOutputNode and doesn't allow duplicate node
+// ids, so we use "root." + source node's plan id as PartitionedOutputNode's.
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:2775]:
```
s/ids, so/ids. Hence/
```

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 4]
```diff
@@ -2824,6 +2834,7 @@ core::PlanFragment VeloxQueryPlanConverterBase::toVeloxQueryPlan(
     }
   }
   auto outputType = toRowType(partitioningScheme.outputLayout, typeParser_);
+  auto partitionedOutputNodeId = toPartitionedOutputNodeId(fragment.root->id);
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:2837]: const auto

**Review by xiaoxmeng — APPROVED** [review]
@kewang1024 LGTM. Thanks!

**Review by zacw7 — APPROVED** [review]

**Pull request merged**